### ISSUE TITLE
filter search instead of prefix search

### ISSUE THREAD

**Issue by aspiers** [rickyrockrat/parcellite#66]
Currently when displaying the list of clipboard items, it is possible to jump to a particular item by typing a unique prefix string. However there are a few possibilities which would significantly improve this feature:

- Rather than just jumping the selection to the first prefix match, instantly filter out all non-matches after each keystroke
- Highlight the matching part of the string in bold text and/or in a different colour.
- Match by any substring, rather than just prefix.

Of course some people may not like all of those, in which case they could be made optional.  But enabling all of those would bring the search in line with the UX for many popular IDEs and dev tools e.g. [fzf](https://github.com/junegunn/fzf).

**Comment by rickyrockrat** [comment]
I agree the search function often leaves me frustrated, and I implemented it. :) That said, you are looking at a dialog so that presents challenges on 'filtering out non-matches' though I like the idea.
The highlight is not a bad idea either, and I really like the any sub-string, since that is exactly what frustrates me, and further this should be pretty low hanging fruit. That said, I have to re-write the whole thing in prep for GTK2 disappearing soon, so may efforts are focused on finding a replacement for GTK that doesn't suck and won't or can't change at the whim of a relatively small group of developers. *sigh* :)

**Comment by aspiers** [comment]
I feel your pain! :stuck_out_tongue_winking_eye: Thanks a lot for the consideration and the reply, and good luck finding a good toolkit!

**Comment by guidomz** [comment]
I would just like to add filtering not only for alphanumerics but also symbols (e.g., #&!).
